File: 137_SingleNumber_II.py
class Solution(object):
    def singleNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        res = 0
        for i in xrange(0, 32):
            cnt = 0
            for num in nums:
                if num & (1 << i) != 0:
                    cnt += 1
            if cnt % 3 == 1:
                res += (1 << i)
        return res

File: 137_SingleNumber_II.cpp
//29.22%
class Solution {
public:
    int singleNumber(vector<int>& nums) {
		int res(0);
		for(int i = 0; i < 32; ++i){
			int cnt(0);
			for(int j = 0; j < nums.size(); ++j){
				if(nums[j] & (1 << i))
					++cnt;
			}
			if(cnt % 3)
				res |= (1 << i);
		}
		return res;		
    }
};

public int singleNumber(int[] A) {
    int ones = 0, twos = 0;
    for(int i = 0; i < A.length; i++){
        ones = (ones ^ A[i]) & ~twos;
        twos = (twos ^ A[i]) & ~ones;
    }
    return ones;
}